packages/lodestar/src/network/reqResp.ts

```diff
@@ -127,46 +113,37 @@ export class ReqResp extends (EventEmitter as IReqEventEmitterClass) implements
     return await this.sendRequest<BeaconBlocksByRootResponse>(peerInfo, Method.BeaconBlocksByRoot, request);
   }
 
-  private async handleRpcRequestStream(
-    peerId: PeerId, method: Method, source: Promise<Buffer | {slice: () => Buffer}>[]): Promise<ResponseChunk[]> {
-    const result: ResponseChunk[] = [];
-    for await (const val of source) {
-      const data = Buffer.isBuffer(val) ? val : val.slice();
-      const response = await this.handleRequest(peerId, method, data);
-      if (!response.err && (method === Method.BeaconBlocksByRange || method === Method.BeaconBlocksByRoot)) {
-        const blocks = response.output as SignedBeaconBlock[];
-        const chunkResponses = blocks.map(block => ({output: block}));
-        result.push(...(chunkResponses));
-      } else {
-        result.push(response as ResponseChunk);
-      }
-    }  
-    return result;
-  }
-
-  private transformResponse(
-    method: Method,
-    source: Promise<{[Symbol.asyncIterator]: () => AsyncIterator<ResponseChunk>}>
-  ): AsyncIterator<Buffer> {
-    const config = this.config;
-    return (async function * () {
-      const sourceVal = await source;
-      for await (const val of sourceVal) {
-        // each yield result will be sent to stream.sink immediately
-        yield encodeResponseChunk(config, method, val);
-      }
-    })();
+  private handleRpcRequests(
```

Comment:
should be renamed `handleRpcRequest` since this is handling a single request.

Reply:
It's handling multiple same method requests from same peer.Like if peer sends two BlocksByRange requests with different request data, this pipe will handle them both

Reply:
I think this is not the intended behavior described in the spec.
> We use ONE stream PER request/response interaction. Streams are closed when the interaction finishes, whether in success or in error.

Reply:
I know, but that is a bit more complicated and I was planning on creatimg a separate issues for those stuff

Reply:
#675 #676 , #677

Reply:
my understanding is we still use ONE stream per request/response interaction with this implementation. Every time client sends request, it dials and then a new stream returned (connection is the same). At the server side, stream is created and closed automatically per request as I don't see any example to close stream in libp2p api.

If we send 2 BlocksByRange request, it'll dial 2 times and 2 separated streams returned (connection is the same) and closed automatically once done. This same method are used across requests but each time its a separate stream.

am I missing something?

Reply:
https://github.com/libp2p/js-libp2p-mplex

Reply:
I think @tuyennhv is right. Even if we're using mplex or some other stream multiplexer, each dial should return a new stream.

Reply:
![image](https://user-images.githubusercontent.com/8836210/75467663-4966f880-598c-11ea-98d3-2c9f0ba76419.png)